### PR TITLE
pb:1872 Added cronjob for quick maintenance run.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What this PR does / why we need it**:
```
pb:1872 Added cronjob for quick maintenance run.
        - made cron job history to one for successful and failure job runs.
```
**Which issue(s) this PR fixes** (optional)
Closes # pb-1872

**Special notes for your reviewer**:
Testing:
```
[root@central-iris-princess-0 ~]# kubectl get cronjobs -n px-backup
NAME                             SCHEDULE       SUSPEND   ACTIVE   LAST SCHEDULE   AGE
full-maintenance-repo-dbfa25d    */10 * * * *   False     0        <none>          33s
quick-maintenance-repo-dbfa25d   */5 * * * *    False     0        <none>          33s
[root@central-iris-princess-0 ~]#
```

```
Every 2.0s: kubectl get pods -n px-backup                                                                                                                                                  Tue Oct  5 05:51:55 2021

NAME                                              READY   STATUS      RESTARTS   AGE
full-maintenance-repo-dbfa25d-1633413000-bzkdc    0/1     Completed   0          111s
px-backup-b6fd6dfcd-mnlm8                         1/1     Running     0          19m
pxc-backup-mongodb-0                              1/1     Running     0          25h
pxc-backup-mongodb-1                              1/1     Running     0          25h
pxc-backup-mongodb-2                              1/1     Running     0          25h
pxcentral-apiserver-74d496b468-hsjp8              1/1     Running     0          25h
pxcentral-backend-6948bdd64c-nncww                1/1     Running     8          25h
pxcentral-frontend-6f89645b78-7brsd               1/1     Running     0          25h
pxcentral-keycloak-0                              1/1     Running     0          25h
pxcentral-keycloak-postgresql-0                   1/1     Running     0          25h
pxcentral-lh-middleware-7c5fcb75d8-zql6v          1/1     Running     0          25h
pxcentral-mysql-0                                 1/1     Running     0          25h
pxcentral-post-install-hook-bprlz                 0/1     Error       0          25h
pxcentral-post-install-hook-bsxd5                 0/1     Error       0          25h
pxcentral-post-install-hook-hhqbb                 0/1     Completed   0          25h
pxcentral-post-install-hook-jc2p8                 0/1     Error       0          25h
quick-maintenance-repo-dbfa25d-1633413000-78m92   0/1     Completed   0          111s
```

### CODE REVIEW

File: pkg/apis/kdmp/v1alpha1/backuplocationmaintenance.go [hunk 1]
```diff
@@ -43,7 +43,8 @@ type BackupLocationMaintenanceSpec struct {
 
 // BackupLocationMaintenanceStatus defines  the status for BackupLocationMaintenance
 type BackupLocationMaintenanceStatus struct {
-	RepoStatus map[string]RepoMaintenanceStatus
+	FullMaintenacneRepoStatus  map[string]RepoMaintenanceStatus
```
Comment (prashanthpx) [inline pkg/apis/kdmp/v1alpha1/backuplocationmaintenance.go:46]: typo Maintenacne, in all places.

File: pkg/executor/kopia/maintenance.go [hunk 2]
```diff
@@ -148,23 +163,36 @@ func runMaintenance() error {
...
-			statusErr := updateBackupLocationMaintenace(kdmp_api.RepoMaintenanceStatusFailed, repo.Name, err.Error())
-			if statusErr != nil {
-				logrus.Warnf("update of status for repo [%v] failed: %v", repo.Name, statusErr)
+		if maintenanceType == fullMaintenanceType {
+			if err := runKopiaMaintenanceExecute(repo); err != nil {
+				errMsg := fmt.Sprintf("maintenance full run command failed for repo [%v]: %v", repo.Name, err)
+				logrus.Errorf("%s: %v", fn, errMsg)
+				statusErr := updateBackupLocationMaintenace(maintenanceType, kdmp_api.RepoMaintenanceStatusFailed, repo.Name, err.Error())
+				if statusErr != nil {
+					logrus.Warnf("update of status for repo [%v] failed: %v", repo.Name, statusErr)
```
Comment (prashanthpx) [inline pkg/executor/kopia/maintenance.go:179]: nit: Can we add a full maintenance sentence here and the same for quick maintenance below.

**Review by prashanthpx — APPROVED** [review]

File: pkg/apis/kdmp/v1alpha1/backuplocationmaintenance.go (hunk 1 above)
Comment (siva-portworx) [reply pkg/apis/kdmp/v1alpha1/backuplocationmaintenance.go]: fixed

File: pkg/executor/kopia/maintenance.go (hunk 2 above)
Comment (siva-portworx) [reply pkg/executor/kopia/maintenance.go]: Fixed

**Pull request merged**